### PR TITLE
Bug 1089154 - [Messages] css refinement (clean up and remove certain pseudo class removal)

### PR DESCRIPTION
**Pull request by steveck-chung**

### CODE REVIEW

**Comment by try-server-hook** [comment]
[steveck-chung steveck-chung](https://github.com/steveck-chung) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=4b6d8ced69f2)

File: apps/sms/style/sms.css [hunk 1]
```diff
@@ -302,7 +300,7 @@ form[role="dialog"][data-type="edit"] .bb-tablist > li {
   background: #fff;
 }
 
-.panel:first-child {
+#thread-list {
```
Comment (azasypkin) [inline apps/sms/style/sms.css]: nit: maybe it would be better to have ```.default-panel``` instead?

File: apps/sms/style/sms.css [hunk 2]
```diff
@@ -890,11 +873,7 @@ form[role="dialog"][data-type="action"].contact-prompt > header {
   -moz-margin-end: -1.5rem; /* compensate the header's 3rem padding */
 }
 
-section[role="region"].new .article-list[data-type="list"] li:first-child {
-  margin-top: 4.6rem;
-}
-
-section[role="region"].new .article-list[data-type="list"]  li {
+#messages-recipient-suggestions li {
```
Comment (azasypkin) [inline apps/sms/style/sms.css]: nit: we have ```#messages-recipient-suggestions``` style above, maybe we can group them together

File: shared/js/option_menu.js [hunk 3]
```diff
@@ -119,9 +119,13 @@ var OptionMenu = function(options) {
   menu.dataset.items = items.length;
 
   // For each option, we append the item and listener
-  items.forEach(function renderOption(item) {
+  items.forEach(function renderOption(item, idx) {
     var button = document.createElement('button');
+
     button.type = 'button';
+    if (idx === items.length -1) {
```
Comment (azasypkin) [inline shared/js/option_menu.js]: nit: ``` - 1```

File: shared/style/action_menu.css [hunk 4]
```diff
@@ -238,7 +238,7 @@
   font-style: italic;
 }
 
-[role="dialog"][data-type="action"] > menu > button:last-child {
+[role="dialog"][data-type="action"] > menu > button.last {
```
Comment (azasypkin) [inline shared/style/action_menu.css]: mmm I'm afraid that there are forms in gaia that use this selector, but don't have ```last``` selector or you have a plan to update every place?

**Comment by autolander** [comment]
Autolander could not find a bug number in your pull request title. All pull requests should be in the format of: Bug [number] - [description].

**Comment by try-server-hook** [comment]
[steveck-chung steveck-chung](https://github.com/steveck-chung) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=3725208a6f5e)

**Comment by try-server-hook** [comment]
[steveck-chung steveck-chung](https://github.com/steveck-chung) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=bd85f69f2964)

File: apps/sms/style/sms.css [hunk 5]
```diff
@@ -315,6 +311,10 @@ form[role="dialog"][data-type="edit"] .bb-tablist > li {
   transition: transform 0.4s ease;
 }
 
+.panel.default {
```
Comment (azasypkin) [inline apps/sms/style/sms.css]: nit: maybe ```.default-panel``` would be less ambiguous?

File: shared/style/edit_mode.css [hunk 6]
```diff
@@ -15,27 +15,20 @@
...
 [role="dialog"][data-type="edit"] [role="region"] > gaia-header h1 {
   color: #868692;
 }
 
-[role="dialog"][data-type="edit"] [role="region"] > header:first-child menu[type="toolbar"] button,
 [role="dialog"][data-type="edit"] [role="region"] > gaia-header button {
   color: #00aac5;
 }
 
-[role="dialog"][data-type="edit"] [role="region"] > header:first-child .icon.icon-close {
```
Comment (azasypkin) [inline shared/style/edit_mode.css]: Looks like all these removed selectors are required only in "example" file [1] and don't have any value for real usages. Probably we'll need to confirm with Pavel if it's ok that [1] is affected :)

[1]  ```/shared/style/edit_mode/index.html```

File: shared/style/edit_mode.css [hunk 7]
```diff
@@ -122,7 +115,3 @@
   background-color: #e51e1e;
   color: #fff;
 }
-
-[role="dialog"][data-type="edit"] > menu button:first-child {
```
Comment (azasypkin) [inline shared/style/edit_mode.css]: mmm looks like this rule is at least required by Contacts [1] or I missing something?

[1] https://github.com/azasypkin/gaia/blob/a6881205deae450757a8d1e1ed65e5e5be0ec633/apps/communications/contacts/index.html#L234

**Comment by try-server-hook** [comment]
[steveck-chung steveck-chung](https://github.com/steveck-chung) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=8993b89f9fc9)

**Comment by try-server-hook** [comment]
[steveck-chung steveck-chung](https://github.com/steveck-chung) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=21c8d9140e9d)

**Comment by try-server-hook** [comment]
[steveck-chung steveck-chung](https://github.com/steveck-chung) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=7a70820b469f)

**Comment by try-server-hook** [comment]
[steveck-chung steveck-chung](https://github.com/steveck-chung) started tests. [Results](https://treeherder.mozilla.org/ui/#/jobs?repo=gaia-try&revision=077caf6ccede)

**Pull request merged**